lib/timers.js

```diff
@@ -481,6 +481,7 @@ Timeout.prototype.unref = function() {
     }
 
     var handle = reuse(this);
+    delete handle._list;
```

Comment:
@Fishrock123 

> delete is known to be a heavy operation.

I thought we were also considering setting the property to `null` or `undefined`, isn't it still the case?

You mentioned earlier:

> Setting it to undefined would be faster but I figured it could be confusing that sometimes there would be a _list property.

> If that is acceptable we can switch to it though.

Having a `_list` property set to `null` or `undefined` seems like it would be acceptable.